cloudferrylib/utils/cache.py

```diff
@@ -30,7 +30,9 @@ def __init__(self, func):
         self.func = func
         self.cache = {}
 
-    def __call__(self, *args):
+    def __call__(self, *args, **kwargs):
+        if kwargs:
+            return self.func(*args, **kwargs)
```

Comment:
As we can see in the docstring of this class, current cache implementation already does not support key-value arguments (without this commit). Moreover, it fails, if we try to use it with method, which uses key-value arguments. And for this actual patch I fixed this specific issue.

So in this case, I suggest implementing this feature (support kwargs in cache) as a separate task with it's own priority, cause it is not related to this patch.